ngraph/core/src/op/unsqueeze.cpp

```diff
@@ -34,23 +34,27 @@ void op::v0::Unsqueeze::validate_and_infer_types()
     const auto data_rank = data_partial_shape.rank();
 
     const auto axes_constant = get_constant_from_source(input_value(1));
+    auto axes_pshape = get_input_partial_shape(1);
+
+    NODE_VALIDATION_CHECK(this,
```

Comment:
Please add type prop test to validate this check, e.g. axes is a Parameter and shape is dynamic